sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/utils/BaseHttpRouterUtils.java

```diff
@@ -110,35 +84,42 @@ public static Optional<ServiceInstance> chooseServiceInstanceByXds(String servic
         }
         List<ServiceInstance> serviceInstanceList = new ArrayList<>(serviceInstanceByXdsRoute);
         XdsLoadBalancer loadBalancer = XdsLoadBalancerFactory
-                .getLoadBalancer(serviceInstanceList.get(0).getClusterName());
+                .getLoadBalancer(serviceName, serviceInstanceList.get(0).getClusterName());
         return Optional.of(loadBalancer.selectInstance(serviceInstanceList));
     }
 
     /**
      * isXdsRouteRequired
      *
-     * @param host host
+     * @param serviceName serviceName
      * @return isXdsRouteRequired
      */
-    public static boolean isXdsRouteRequired(String host) {
-        // if host is ip, so no xds routing required
-        if (StringUtils.isEmpty(host) || host.equals(LOCAL_HOST) || IP_PATTERN.matcher(host).matches()) {
+    public static boolean isXdsRouteRequired(String serviceName) {
+        // if service is localhost or started not with lowercase, so no xds routing required
+        if (StringUtils.isEmpty(serviceName) || serviceName.equals(LOCAL_HOST)
+                || !isStartWithLowercase(serviceName)) {
             return false;
         }
         return true;
     }
 
+    private static boolean isStartWithLowercase(String serviceName) {
+        char firstChar = serviceName.charAt(0);
+        return firstChar >= 'a' && firstChar <= 'z';
```

Comment:
Maybe you can use Character.isLowerCase.

Reply:
done